Hybrid VMs at large scale improvements

afcollins opened the pull request:
Enables hugepages on the hypervisor and libvirt VMs.
Fixes some issues with hugetlb pages.

Also, adds interfaces that can be used as virtual functions.

Some changes were generated using Cursor and the claude-4-sonnet model.

mcornea commented on docs/troubleshooting.md:
````diff
@@ -274,6 +275,15 @@ racadm>>set iDRAC.VirtualMedia.Attached Attached
 Object value modified successfully
 ```
 
+## Failing ImagePull due to Pull Secret
````
Is this scenario for when the user can no longer update the pull-secret by `oc -n openshift-config edit secret/pull-secret` ?

afcollins replied:
Yes. I only encountered it when the credentials installed in the cluster belonged to an account that was deactivated and no nodes would pull images.

I didn't like this section of the Troubleshooting doc to place it under, but wasn't sure there was any better option. Open to suggestions on that too.

afcollins commented on ansible/roles/hv-vm-create/templates/kvm-def.xml.j2:
```diff
@@ -125,6 +133,18 @@
 {% endif %}
       <address type='pci' domain='0x0000' bus='0x01' slot='0x00' function='0x0'/>
     </interface>
+{% for i in range(1, 6) %}
```
Added variable

afcollins commented:
Addressed all comments. Reverted some unnecessary changes. Renamed a variable. Added a variable for VFs in VMs. Please take another look when you can.

I still don't have a way to validate these changes for right now. :/ But may be able to do so in early December.

afcollins commented:
Changing to draft while I split into more atomic PRs.

afcollins commented on ansible/roles/hv-vm-create/tasks/main.yml:
```diff
@@ -20,6 +20,34 @@
+        # TODO This grep expression failed on the second installation. Need either a way to force huge pages, or advanced logic to check which is configured.
+    - name: Check host hugepages availability
+      shell: |
+        grep -E "HugePages_Free.*{{ vm_hugepage_size }}" /proc/meminfo | awk '{print $2}' || echo "0"
```
This failed on my latest installation because the machines only had 1Gi hugepages defined, so only `HugePages_Free` showed up in meminfo. I believe it only suffixes `-1Gi` when both size hugepages are configured.

mcornea replied:
Yep, it sounds reasonable to me.

mcornea commented:
Hi @afcollins Is this PR ready for merge?

mcornea commented:
Awesome, thanks @afcollins  !

mcornea commented:
/lgtm
